### ISSUE TITLE
Training Code

### ISSUE THREAD

**Issue by yimingz1218** [LuoRadisher/CPFlow#1]
Hello, thanks for your excellent open-source code. May I ask when will you release your training code? I would like to tune the model but have a little problem regarding the batched training issue. How should I set eval_basis, sample_guide, norm_time with batch size > 1?

**Comment by LuoRadisher** [comment]
Glad to receive your feedback. The training code has been updated. As for your issue,  the mini-batch clips share the same length so we set a batch sampler to aggregate these samples by their length. The sample_guide and norm_time denote inputs(source) timestamps and output(target) timestamps respectively and normalized by the batch length. The eval_basis denotes the weights matrix (m * n; m for the length of norm_time; n for the num of control points)  of the b-spline curve. The implementation of this function includes a few params(u: norm_time, i: control point idx, k: the fix num for curve degree; nodeVector: a node list ) More theoretical details can be found in our paper and references. I Hope that these explanations help you solve the problem.
